src/model/questionGenerator.js

```diff
@@ -0,0 +1,76 @@
+import { randomNumbers, randomValueFromArray } from "./utilities/utilities";
+
+const generateQuestion = async (type, askedQuestion) => {
+  const data = await fetchData(type);
+  let answersIndexes;
+  let correctAnswerIndex;
+  let areAnswersDifferent;
+  do{
+    answersIndexes = randomNumbers(0, data.length - 1, 4);
+    correctAnswerIndex = randomValueFromArray(answersIndexes);
+    const filtredAnswers = data.filter((_, index) => answersIndexes.includes(index));
+    areAnswersDifferent = areAllDifferent(filtredAnswers);
+  } while(!areAnswersDifferent || askedQuestion.includes(correctAnswerIndex));
+
+  const dataForQuestion = {
+    answersIndexes,
+    correctAnswerIndex,
+    data,
+  };
+
+  return getQuestion(dataForQuestion);
+}
+
+const fetchData = async (type) => {
+  let response = await fetch(`https://breakingbadapi.com/api/${type}`);
+  let data = await response.json();
+  if(type === 'characters'){
+    return charactersTypeData(data);
+    }
+  if(type === 'quotes') {
+    return quoteTypeData(data);
+  }
+  if(type === 'deaths') {
+    return deathTypeData(data);
+  }
+}
+
+const charactersTypeData = (data) => {
+  return data.map(({name, img}) => ({
+    answer: name,
+    questionObject: img,
+  }));
+}
+
+const quoteTypeData = (data) => {
+  return data.map(({quote, author}) => ({
+    answer: author,
+    questionObject: quote,
+  }));
+}
+
+const deathTypeData = (data) => {
+  return data.map(({death, cause})=> ({
+    answer: death,
+    questionObject: cause,
+  }));
+}
+
+const areAllDifferent = (filtredAnswers) => {
+  const answers = [...filtredAnswers.map(({answer})=> answer)];
+  while(answers.length != 1){
+    const answer = answers.pop();
+    if(answers.includes(answer)) {
+      return false;
+    }
+  }
+  return true;
+}
+
+const getQuestion = ( {answersIndexes, correctAnswerIndex, data}) => ({
```

Comment:
```suggestion
const questionObject = ( {answersIndexes, correctAnswerIndex, data}) => ({
```

Nie jest konieczne dawanie `get` - raczej to zła praktyka

Reply:
A najlepiej nie eksportuj do osobnej funkcji. 
Dodaj go w tej linii bezpośrednio!

![image](https://user-images.githubusercontent.com/41823689/144719304-25cbfa2d-7ca0-4b25-b8a0-e8d7e27e8873.png)